exasol_sagemaker_extension/lua/src/install_autopilot_prediction_udf.lua

```diff
@@ -0,0 +1,108 @@
+---
+-- @module install_autopilot_prediction_udf
+--
+-- This module generates and deploys Autopilot prediction udf
+--
+
+local M = {}
+
+local exaerror = require("exaerror")
+
+_G.global_env = {
+	pquery = pquery,
+	error = error
+}
+
+---
+-- Split the given string using the specified seperator into a list of strings
+--
+-- @param character		The separator to use when splitting the string.
+--
+-- @return  a list of strings
+--
+function string:split(seperator)
+  local splitted_result = {}
+  local i = 1
+  for str in string.gmatch(self, "[^"..seperator.."]+") do
+    splitted_result[i] = str
+    i = i + 1
+  end
+  return splitted_result
+end
+
+---
+-- Generate and deploy prediction udf
+--
+-- @param schema		       	The name of schema where the prediction udf gets created
+-- @param endpoint_name     	The name of endpoint is for which prediction UDF is created
+-- @param model_conn_name   	The name of connection holds information about mode
+-- @param input_params          Input parameters for the prediction udf
+-- @param output_params			Output for the prediction udf
+--
+function M.install_udf(schema, endpoint_name, model_conn_name, input_params, output_params)
+    local query_create  =
+	"CREATE OR REPLACE PYTHON3_SME SET SCRIPT " .. schema .. ".SME_ENDPOINT_PREDICTION_" .. endpoint_name .. "_UDF("
+	.. table.concat(input_params, ',')  .. ") EMITS (" .. table.concat(output_params, ',') .. ") AS\n"
+	..
+	"from exasol_sagemaker_extension.autopilot_prediction import AutopilotPredictionUDF\n"
+	..
+	"udf = AutopilotPredictionUDF(exa, '" .. model_conn_name .. "')\ndef run(ctx):\n\tudf.run(ctx)\n/"
+
+    local success, result = _G.global_env.pquery(query_create)
```

Comment:
It took me a while to realize that where I defined a python function with a colon, I shouldn't be using the query parameter.

Reply:
yeah, this code is a bit confusing. Not sure, if it is possible, to somehow to inject the UDF code while deployment.

Reply:
Since indentation is matter in python, string concatenation seem safer method to produce correct python code.